core/components/atoms/input/__tests__/Input.test.tsx

```diff
@@ -209,3 +209,773 @@ describe('Input component selection range is set correctly on input type change'
+
+      const input = getByTestId('DesignSystem-Input');
+      fireEvent.click(input);
+
+      expect(onClickMock).toHaveBeenCalledTimes(1);
+    });
+
+    it('calls onPaste handler when content is pasted', () => {
+      const onPasteMock = jest.fn();
+      const { getByTestId } = render(<Input name="test" onPaste={onPasteMock} />);
+
+      const input = getByTestId('DesignSystem-Input');
+      fireEvent.paste(input, { clipboardData: { getData: () => 'pasted text' } });
+
+      expect(onPasteMock).toHaveBeenCalledTimes(1);
+    });
+
+    it('focuses input when wrapper is clicked', () => {
+      const { getByTestId } = render(<Input name="test" />);
+
+      const input = getByTestId('DesignSystem-Input') as HTMLInputElement;
+      const wrapper = getByTestId('DesignSystem-InputWrapper');
+
+      input.focus = jest.fn();
+
+      fireEvent.click(wrapper);
+      expect(input.focus).toHaveBeenCalled();
+    });
+  });
+
+  describe('Clear functionality', () => {
+    it('shows clear icon when onClear is provided and input has value', () => {
+      const onClearMock = jest.fn();
+      const { getByTestId } = render(<Input name="test" value="test value" onClear={onClearMock} />);
+
+      const clearIcon = getByTestId('DesignSystem-Input--closeIcon');
+      expect(clearIcon).toBeInTheDocument();
+    });
+
+    it('does not show clear icon when input is empty', () => {
+      const onClearMock = jest.fn();
+      const { queryByTestId } = render(<Input name="test" value="" onClear={onClearMock} />);
+
+      const clearIcon = queryByTestId('DesignSystem-Input--closeIcon');
+      expect(clearIcon).not.toBeInTheDocument();
+    });
+
+    it('calls onClear when clear icon is clicked', () => {
+      const onClearMock = jest.fn();
+      const { getByTestId } = render(<Input name="test" value="test value" onClear={onClearMock} />);
+
+      const clearIcon = getByTestId('DesignSystem-Input--closeIcon');
+      fireEvent.click(clearIcon);
+
+      expect(onClearMock).toHaveBeenCalledTimes(1);
+    });
+
+    it('focuses input after clear icon is clicked', () => {
+      const onClearMock = jest.fn();
+      const { getByTestId } = render(<Input name="test" value="test value" onClear={onClearMock} />);
+
+      const input = getByTestId('DesignSystem-Input') as HTMLInputElement;
+      const clearIcon = getByTestId('DesignSystem-Input--closeIcon');
+
+      input.focus = jest.fn();
+
+      fireEvent.click(clearIcon);
+      expect(input.focus).toHaveBeenCalledWith({ preventScroll: true });
+    });
+
+    it('shows clear icon with defaultValue when onClear is provided', () => {
+      const onClearMock = jest.fn();
+      const { getByTestId } = render(<Input name="test" defaultValue="default value" onClear={onClearMock} />);
+
+      const clearIcon = getByTestId('DesignSystem-Input--closeIcon');
+      expect(clearIcon).toBeInTheDocument();
+    });
+  });
+
+  describe('Action icon functionality', () => {
```

Comment:
Please write proper description for all the test cases. Also specify the **"Input Component"** in the description to clearly identify that this test case belongs to which component.